rcl/src/rcl/publisher.c

```diff
@@ -437,9 +438,19 @@ rcl_publisher_get_actual_qos(const rcl_publisher_t * publisher)
 bool
 rcl_publisher_can_loan_messages(const rcl_publisher_t * publisher)
 {
+  const char * env_val = NULL;
+
   if (!rcl_publisher_is_valid(publisher)) {
     return false;  // error message already set
   }
+  const char * env_error_str = rcutils_get_env(RCL_DISABLE_LOAN_MSG_ENV_VAR, &env_val);
+  if (NULL != env_error_str) {
+    RCUTILS_SAFE_FWRITE_TO_STDERR("Error getting env var: ");
+    RCUTILS_SAFE_FWRITE_TO_STDERR(RCL_DISABLE_LOAN_MSG_ENV_VAR);
+    RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
+  } else if (env_val != NULL && strcmp(env_val, "1") == 0) {
+    return false;
+  }
```

Comment:
This code could be factored out into a separate file so this logic is not repeated in the publisher and subscription functions.

Reply:
true, move this into `rcl/include/rcl/node.h`.